rllib/env/base_env.py

```diff
@@ -264,14 +339,16 @@ def _space_contains(space: gym.Space, x: MultiEnvDict) -> bool:
         """
         # this removes the agent_id key and inner dicts
         # in MultiEnvDicts
-        flattened_obs = {
-            env_id: list(obs.values())
-            for env_id, obs in x.items()
-        }
+        agents = set(self.get_agent_ids())
         ret = True
-        for env_id in flattened_obs:
-            for obs in flattened_obs[env_id]:
-                ret = ret and space[env_id].contains(obs)
+        for _, multi_agent_dict in x.items():
```

Comment:
if you are ignoring keys, why not x.values()?

Reply:
fixed